src/createAgentManager.ts

```diff
@@ -291,6 +291,8 @@ export async function createAgentManager(agent: string, options: AgentManagerOpt
     };
 
     async function connect(newChat: boolean) {
+        options.callbacks.onConnectionStateChange?.(ConnectionState.Connecting);
```

Comment:
@HaMMeRSI is it safe to set this on our own, even when ice does it? can the double assignment cause issues?